posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -1508,4 +1543,26 @@ async def run(self, inputs: RunWorkflowInputs) -> Results:
             ),
         )
 
+        if (
+            settings.DUCKLAKE_DATA_MODELING_COPY_WORKFLOW_ENABLED
+            and self.ducklake_copy_inputs
+            and self.ducklake_copy_inputs.models
+        ):
+            temporalio.workflow.logger.info(
+                "Triggering DuckLake copy child workflow",
+                job_id=job_id,
+                models=len(self.ducklake_copy_inputs.models),
+            )
+            await temporalio.workflow.start_child_workflow(
+                workflow="ducklake-copy.data-modeling",
+                arg=dataclasses.asdict(self.ducklake_copy_inputs),
+                id=f"ducklake-copy-data-modeling-{job_id}",
+                task_queue=settings.DATA_MODELING_TASK_QUEUE,
+                parent_close_policy=ParentClosePolicy.ABANDON,
+                retry_policy=temporalio.common.RetryPolicy(
+                    maximum_attempts=1,
+                    non_retryable_error_types=["NondeterminismError"],
+                ),
```

Comment:
Because `start_child_workflow` is a `fire and forget` method, when using `file_uris=delta_table.file_uris()` to get the parquet files, the materialization job could run again and modify the files before the child workflow has fully copied them over - a bit of an (unlikely) race condition.

What we do is copy the delta files to a new "queryable" folder for clickhouse to hit. We keep multiple versions of this folder so that any in-progress clickhouse queries dont fail when materialization runs again. These query folders are kept around for a minimum of 10 mins before a future materialization job is allowed to delete them. Check out the `queryable_folder` column on `DataWarehouseTable` - this is the extra folder path that you can use to copy the files using duckdb

Reply:
Depending on how duckdb does schema scans on parquet files, this may be a smart move. We've had problems with clickhouse only reading the first parquet file to get the schema which can cause schema drift when other parquet files include extra columns. 

So yeah, maybe going back to a delta scan is actually good here - sorry for my noise on this!